lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -98,7 +116,80 @@ class MockTimers {
     this.#executionQueue.removeAt(position);
   }
 
-  async * #setIntervalPromisified(interval, startTime, options) {
+  #createDate() {
+    kMock ??= Symbol('MockTimers');
+    /**
+     * Function to mock the Date constructor, treats cases as per ECMA-262
+     * and returns a Date object with a mocked implementation
+     * @typedef {Date} MockDate
+     * @returns {MockDate} a mocked Date object
+     */
+    function MockDate(year, month, date, hours, minutes, seconds, ms) {
+      const mockTimersSource = MockDate[kMock];
+      const nativeDate = mockTimersSource.#RealDate;
+      // As of the fake-timers implementation for Sinon
+      // ref https://github.com/sinonjs/fake-timers/blob/a4c757f80840829e45e0852ea1b17d87a998388e/src/fake-timers-src.js#L456
+      // This covers the Date constructor called as a function ref.
+      // ECMA-262 Edition 5.1 section 15.9.2.
+      // and ECMA-262 Edition 14 Section 21.4.2.1
+      if (!(this instanceof MockDate)) {
+        return DatePrototypeToString(new nativeDate(mockTimersSource.#now));
+      }
+
+      // Cases where Date is called as a constructor
+      // This is intended as a defensive implementation to avoid
+      // having unexpected returns
+      switch (arguments.length) {
+        case 0:
+          return new nativeDate(MockDate[kMock].#now);
+        case 1:
+          return new nativeDate(year);
+        case 2:
+          return new nativeDate(year, month);
+        case 3:
+          return new nativeDate(year, month, date);
+        case 4:
+          return new nativeDate(year, month, date, hours);
+        case 5:
+          return new nativeDate(year, month, date, hours, minutes);
+        case 6:
+          return new nativeDate(year, month, date, hours, minutes, seconds);
+        default:
+          return new nativeDate(year, month, date, hours, minutes, seconds, ms);
+      }
+    }
+
+    // Prototype is read-only, and non assignable through Object.defineProperties
+    // eslint-disable-next-line no-unused-vars -- used to get the prototype out of the object
+    const { prototype, ...dateProps } = ObjectGetOwnPropertyDescriptors(this.#RealDate);
+
+    // Binds all the properties of Date to the MockDate function
+    ObjectDefineProperties(
+      MockDate,
+      dateProps,
+    );
+
+    MockDate.now = function now() {
+      return MockDate[kMock].#now;
+    };
+
+    // This is just to print the function { native code } in the console
+    // when the user prints the function and not the internal code
+    MockDate.toString = function toString() {
+      return FunctionPrototypeToString(MockDate[kMock].#RealDate);
+    };
+
+    MockDate.prototype = this.#RealDate.prototype;
+    MockDate.parse = this.#RealDate.parse;
+    MockDate.UTC = this.#RealDate.UTC;
+    MockDate.prototype.toUTCString = this.#RealDate.prototype.toUTCString;
+    MockDate.isMock = true;
+    MockDate[kMock] = this;
```

Comment:
this makes both `isMock` and `kMock` enumerable, which means Object.assign and object spread will copy them. I think you want to use `ObjectDefineProperty` here to make them non-enumerable.

Reply:
`isMock` I think it should be enumerable to let users know when they're dealing with mocks, what do you think?